src/Utils.elm

```diff
@@ -0,0 +1,13 @@
+module Utils exposing
+    ( toggleList
+    )
+
+
+toggleList : 
+    List a
+    -> a 
+    -> List a
+toggleList list item =
+    if List.member item list
+    then List.filter (\x -> x /= item) list
+    else List.append list [item]
```

Comment:
Assuming we don't care about order, this would be much better:

```suggestion
    else item :: list
```

Otherwise it would be far better to use `Set` for data we need to perform these operations with. Set itself is contained to `comparable` types but I wrote [any-set](https://package.elm-lang.org/packages/turboMaCk/any-set/latest/) to solve this.